bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/types/StateDescriptionFragmentBuilder.java

```diff
@@ -0,0 +1,239 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.core.types;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * Builds a {@link StateDescriptionFragment} with the relevant parts only.
+ *
+ * @author Henning Treu - initial contribution and API.
+ *
+ */
+@NonNullByDefault
+public class StateDescriptionFragmentBuilder {
+
+    private @Nullable BigDecimal minimum;
+    private @Nullable BigDecimal maximum;
+    private @Nullable BigDecimal step;
+    private @Nullable String pattern;
+    private @Nullable Boolean readOnly;
+    private @Nullable List<StateOption> options;
+
+    /**
+     * Return a fresh builder instance.
+     *
+     * @return a fresh {@link StateDescriptionFragmentBuilder} instance.
+     */
+    public static StateDescriptionFragmentBuilder instance() {
```

Comment:
Maybe it's personal bias, but to me the name `instance()` implies that I will get the singleton instance of it. We have `ThingBuilder.create(...)` and `ChannelBuilder.create(...)`. Admittedly I overlooked  `ThingTypeBuilder.instance(...)`, so it became inconsistent already, but my personal preference would be to keep it as consistent as possible and stay with the majority...

Reply:
agreed.